src/configuration/Configuration.cpp

```diff
@@ -161,6 +177,22 @@ namespace SlimeVR {
             m_Calibrations[sensorID] = config;
         }
 
+        size_t Configuration::getWiFiCredentialCount() const {
+            return m_WiFiCredentials.size();
+        }
+
+        const std::map<const char*, WiFiCredential>* const Configuration::getWiFiCredentials() const {
+            return &m_WiFiCredentials;
+        }
+
+        void Configuration::setWiFiCredential(const WiFiCredential& credential) {
```

Comment:
This seems not to insert a new element. 
 m_Logger.trace("m_WiFiCredentials size: %d", m_WiFiCredentials.size()); returns each time 1
```
[TRACE] [Configuration] Found WiFi credential file: /wifi_credentials/test1
[INFO ] [Configuration] Found up-to-date WiFi credential v1
[TRACE] [Configuration] m_WiFiCredentials size: 1
[TRACE] [Configuration] Found WiFi credential file: /wifi_credentials/test2
[INFO ] [Configuration] Found up-to-date WiFi credential v1
[TRACE] [Configuration] m_WiFiCredentials size: 1
[TRACE] [Configuration] Found WiFi credential file: /wifi_credentials/test3
[INFO ] [Configuration] Found up-to-date WiFi credential v1
[TRACE] [Configuration] m_WiFiCredentials size: 1
[TRACE] [Configuration] Found WiFi credential file: /wifi_credentials/test4
[INFO ] [Configuration] Found up-to-date WiFi credential v1
[TRACE] [Configuration] m_WiFiCredentials size: 1
------
[TRACE] [StatusManager] Added status WIFI_CONNECTING
[TRACE] [WiFiHandler] Adding WiFi credential: test4
[INFO ] [WiFiHandler] Loaded credentials for "test4"
```

Reply:
This is probably why: https://stackoverflow.com/a/4157729

Reply:
Confirmed, all the credential ssids have the same memory address, so they all have the same key.

Reply:
m_Logger.trace("m_WiFiCredentials size: %d, SSID: %s, %p", m_WiFiCredentials.size(), credential.ssid, &credential.ssid);
```
[TRACE] [Configuration] Found WiFi credential file: /wifi_credentials/test1
[INFO ] [Configuration] Found up-to-date WiFi credential v1
[TRACE] [Configuration] m_WiFiCredentials size: 1, SSID: fentozell, 0x3ffb1eb8
[TRACE] [Configuration] Found WiFi credential file: /wifi_credentials/test2
[INFO ] [Configuration] Found up-to-date WiFi credential v1
[TRACE] [Configuration] m_WiFiCredentials size: 1, SSID: test2, 0x3ffb1eb8
[TRACE] [Configuration] Found WiFi credential file: /wifi_credentials/test3
[INFO ] [Configuration] Found up-to-date WiFi credential v1
[TRACE] [Configuration] m_WiFiCredentials size: 1, SSID: test3, 0x3ffb1eb8
[TRACE] [Configuration] Found WiFi credential file: /wifi_credentials/test4
[INFO ] [Configuration] Found up-to-date WiFi credential v1
[TRACE] [Configuration] m_WiFiCredentials size: 1, SSID: test4, 0x3ffb1eb8
```

Reply:
Should be fixed, please try again